### DISCUSSION TITLE
Problem to access PMD workflow action, looks like it is IP problem because of the way my company works

### DISCUSSION THREAD

**Discussion by ddg4** [pmd/pmd#4111] [General]
Hello all, I am trying to use PMD workflow actions to analyze Salesfoce Apex code, our company is very restrictive so they have a whitelist and they approved me to have the PMD workflow, but after they add it in the whitelist I am having the following issue:

"Although you appear to have the correct authorization credentials, the `Equifax` organization has an IP allow list enabled, and 20.225.180.211 is not permitted to access this resource." They can give me access to that ip but the ip is not always the same one, so they ask me if PMD has a range ips to access that they can allow, is this something you guys can help me?

Thanks in advance.

@adangel

**Comment by jsotuyod** [comment]
Github actions run on github's infra, PMD is not involved at all, we simply publish the scripts.

The docs for github's ip range are available here https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#ip-addresses

**Reply by ddg4** [reply]
Thanks for your answer @jsotuyod 

What about this github action, is not manage by you guys?
https://github.com/marketplace/actions/pmd

**Reply by jsotuyod** [reply]
Yes, we manage and publish it, but it is run on github's infra as a github action.

**Reply by ddg4** [reply]
Ok I understand, thanks for the URL you share with me, I will talk with my team about it, appreciate it.

Just my last question, that Github action allow to analyze apex code as well, right?

**Reply by adangel** [reply]
> Just my last question, that Github action allow to analyze apex code as well, right?

Yes, it works with Apex code as well

**Reply by ddg4** [reply]
Thanks you guys, I appreciate your help :D
